Doc/library/pathlib.rst

```diff
@@ -946,6 +946,108 @@ call fails (for example because the path doesn't exist).
    to the directory after creating the iterator, whether a path object for
    that file be included is unspecified.
 
+.. method:: Path.walk(top_down=True, on_error=None, follow_symlinks=False)
+
+   Generate the file names in a directory tree by walking the tree
+   either top-down or bottom-up.
+
+   For each directory in the directory tree rooted at *self* (including
+   *self* but excluding '.' and '..'), yields a 3-tuple
+   ``(dirpath, dirnames, filenames)``
+
+   *dirpath* is a :class:`Path` to the directory, *dirnames* is a list of the names
+   of the subdirectories in *dirpath* (excluding ``'.'`` and ``'..'``), and
+   *filenames* is a list of the names of the non-directory files in *dirpath*.
+   Note that the names in the lists contain no path components. To get a full
+   path (which begins with *self*) to a file or directory in *dirpath*, do
+   ``dirpath / name``. Whether or not the lists are sorted depends on the file
+   system.
+
+   If optional argument *top_down* is ``True`` or not specified, the triple for a
+   directory is generated before the triples for any of its subdirectories
+   (directories are generated top-down).  If *top_down* is ``False``, the triple
+   for a directory is generated after the triples for all of its subdirectories
+   (directories are generated bottom-up). No matter the value of *top_down*, the
+   list of subdirectories is retrieved before the tuples for the directory and
+   its subdirectories are generated.
+
+   When *top_down* is True, the caller can modify the *dirnames* list in-place
+   (For example, using :keyword:`del` or slice assignment), and :meth:`Path.walk`
+   will only recurse into the subdirectories whose names remain in *dirnames*;
+   this can be used to prune the search, or to impose a specific order of visiting,
+   or even to inform :meth:`Path.walk` about directories the caller creates or
+   renames before it resumes :meth:`Path.walk` again. Modifying *dirnames* when
+   *top_down* is False has no effect on the behavior of :meth:`Path.walk()`, since the
+   directories in *dirnames* have already been generated by the time *dirnames*
+   is yielded to the caller.
+
+   By default errors from :func:`os.scandir` call are ignored.  If
+   optional argument *on_error* is specified, it should be a callable; it
+   will be called with one argument, an :exc:`OSError` instance. It can
+   report the error to continue with the walk, or raise the exception
```

Comment:
What do you mean by "report the error to continue the walk"? Do you mean suppress/consume the exception, or else re-raise it?

Reply:
Yes, exactly that.